frontend/components/Icon/Icon.tsx

```diff
@@ -5,23 +5,28 @@ import classnames from "classnames";
 interface IIconProps {
   name: IconNames;
   color?: string;
+  direction?: "up" | "down" | "left" | "right";
   className?: string;
 }
 
 const baseClass = "icon";
 
-const Icon = ({ name, color, className }: IIconProps) => {
+const Icon = ({ name, color, direction, className }: IIconProps) => {
   const classsNames = classnames(baseClass, className);
 
   // createPassedProps creates a props object that we pass to the specific icon
   // for values that are not null or undefined
   const props = useMemo(() => {
     const createPassedProps = () => {
-      return Object.assign({}, color === undefined ? undefined : { color });
+      // TODO: how to change color AND direction
+      return Object.assign(
```

Comment:
Is this what you need for the todo?

```
const pasedProps: Record<string, string> = {}
Object.entries(props).forEach(([k, v]) => k[v] !== null &&  k[v] !== undefined && passedProps[k] = v)
return passedProps
```

Reply:
If there are some IIconProps that are defined but that you don't want to pass, you could add something like this before the return:

```
PROPS_TO_EXCLUDE = ["className", "someOtherPropToExclude", "yetAnotherPropToExclude"];
PROPS_TO_EXCLUDE.forEach((propName) => delete passedProps[propName];
```

Reply:
Or maybe this:

```
const OPTIONS = ["color", "direction"] as const; // more options can be added in the future
const passedProps = useMemo(() => {
    const result: IIconComponentProps = {};
    OPTIONS.forEach((optionName) => {
        props[name] !== null && props[name] !== undefined && result[name] = props[name]);
    });
    return result;
}, [props]);
```

Reply:
Sorry for all the comments! My javascript brain got nerd-sniped 🙃

Reply:
@gillespi314 love it. I made a solution, let me know whatcha think <3